### ISSUE TITLE
Line dashes on Windows 200% display scale (hi-dpi) are half-size

### ISSUE THREAD

**Issue by Phillipus** [eclipse-platform/eclipse.platform.swt#687]
This affects Windows only, the problem is not seen on Mac or Linux.

Dashes on lines drawn on an SWT `GC` using `setLineDash` are half-size on Windows 200% scaling.

- Windows 10 or 11
- Java 17
- Any version of SWT

Snippet:

```java
import org.eclipse.swt.graphics.Color;
import org.eclipse.swt.layout.FillLayout;
import org.eclipse.swt.widgets.Display;
import org.eclipse.swt.widgets.Shell;

public class LineDashesSWT {

    public static void main(String[] args) {
        Display display = new Display();
        Shell shell = new Shell(display);
        shell.setLayout(new FillLayout());
        shell.setSize(220, 180);
        shell.setBackground(new Color(null, 255, 255, 255));

        
        shell.addPaintListener(e -> {
            e.gc.setLineDash(new int[] { 3 });
            e.gc.drawRectangle(10, 10, 180, 80);
            e.gc.drawLine(10, 50, 190, 50);
        });
        
        shell.open();

        while(!shell.isDisposed()) {
            if(!display.readAndDispatch()) {
                display.sleep();
            }
        }
    }
}
```
On Windows 100% scaling (this is how it should render):

![win1x](https://github.com/eclipse-platform/eclipse.platform.swt/assets/600504/7c0afa73-2fc0-45d8-93e0-dcf6895d965d)

On Windows 200% scaling:

![win2x](https://github.com/eclipse-platform/eclipse.platform.swt/assets/600504/36c95526-061d-4cb6-b0df-3d9c8be51132)

Compare with Mac and Linux:

Linux 100%:

![linux1x](https://github.com/eclipse-platform/eclipse.platform.swt/assets/600504/b71db41a-3009-499d-8210-bd4ab70232c5)

Linux 200%:

![linux2x](https://github.com/eclipse-platform/eclipse.platform.swt/assets/600504/1ea88e53-a2b7-4b08-a6cc-cfd4ea893161)

Mac 100%:

![mac1x](https://github.com/eclipse-platform/eclipse.platform.swt/assets/600504/602e3ba0-03e7-494c-872b-abb2b55e9e18)

Mac 200%:

![mac2x](https://github.com/eclipse-platform/eclipse.platform.swt/assets/600504/4e11574c-7ed3-4c10-a97b-5e04a54425af)

**Comment by deepika-u** [comment]
@Phillipus 

Seeing like this at 100%
![image](https://github.com/eclipse-platform/eclipse.platform.swt/assets/46807064/f7056344-20c8-4ddb-b5e5-0b980adf70d6)

Seeing like this at 200%
![image](https://github.com/eclipse-platform/eclipse.platform.swt/assets/46807064/9ed024c9-5314-48a8-ac2c-b93667078d9b)

I tried on windows 11, so seeing the issue as you reported.

Eclipse SDK
Version: 2023-06 (4.28)
Build id: I20230605-0440
OS: Windows 11, v.10.0, x86_64 / win32
Java vendor: Eclipse Adoptium
Java runtime version: 20+36
Java version: 20

**Comment by Phillipus** [comment]
PR: https://github.com/eclipse-platform/eclipse.platform.swt/pull/781

**Comment by HeikoKlare** [comment]
fixed via #781
